react/features/device-selection/functions.js

```diff
@@ -20,10 +20,10 @@ export function getDeviceSelectionDialogProps(stateful: Object | Function) {
             !JitsiMeetJS.isMultipleAudioInputSupported(),
         disableDeviceChange:
             !JitsiMeetJS.mediaDevices.isDeviceChangeAvailable(),
-        hasAudioPermission: JitsiMeetJS.mediaDevices
-            .isDevicePermissionGranted.bind(null, 'audio'),
-        hasVideoPermission: JitsiMeetJS.mediaDevices
-            .isDevicePermissionGranted.bind(null, 'video'),
+        hasAudioPermission: () => JitsiMeetJS.mediaDevices
+            .isDevicePermissionGranted('audio'),
+        hasVideoPermission: () => JitsiMeetJS.mediaDevices
```

Comment:
I forget why this needs to be a passed in function. I'm sure there are already a bunch of unnecessary re-renders but passing new functions as props would trigger more re-renders. Do you have insight?

Reply:
Not really. I passed the promises since the existing code expected a function. TBH I'm not sure how to better do this.